kpet/data.py

```diff
@@ -203,34 +205,66 @@ def __init__(self, data):
 
 class Case(Object):     # pylint: disable=too-few-public-methods
     """Test case"""
+
     def __init__(self, data):
+        def convert(old_data):
+            """Convert old data to new data."""
+            new_data = old_data.copy()
+
+            if 'task_params' in old_data.keys():
+                del new_data['task_params']
+                new_data['environment'] = old_data['task_params']
+
+            return new_data
+
         super().__init__(
-            Struct(
-                required=dict(
-                    name=String(),
-                    max_duration_seconds=Int(),
+            Succession(
+                Struct(
+                    required=dict(
+                        name=String(),
+                        max_duration_seconds=Int(),
+                    ),
+                    optional=dict(
+                        host_type_regex=Regex(),
+                        hostRequires=String(),
+                        partitions=String(),
+                        kickstart=String(),
+                        match=Class(PositivePattern),
+                        dont_match=Class(NegativePattern),
+                        waived=Boolean(),
+                        role=String(),
+                        url_suffix=String(),
+                        task_params=Dict(String()),
+                    )
+                ),
+                convert,
+                Struct(
+                    required=dict(
+                        name=String(),
+                        max_duration_seconds=Int(),
+                    ),
+                    optional=dict(
+                        host_type_regex=Regex(),
+                        hostRequires=String(),
+                        partitions=String(),
+                        kickstart=String(),
+                        match=Class(PositivePattern),
+                        dont_match=Class(NegativePattern),
+                        waived=Boolean(),
+                        role=String(),
+                        url_suffix=String(),
+                        environment=Dict(String()),
+                    )
```

Comment:
Looking at this, it might be better to just add support for aliases to `Struct`. E.g. have it accept the optional `aliases` argument, and let that be a string-string (alias-canonical name) dictionary, where keys must not be present in either `required` or `optional` and values must be. Then update `validate` to recognize aliases as the canonical fields, making sure both are not present, and update `resolve` to rename aliases to canonical fields.

What do you think?

Although, it is of course faster to do this instead, at this moment, we might save time on future renames.

Otherwise this looks OK. Only maybe put down a TODO to remove the succession once the database is updated.

Reply:
Another (probably simplest) way could be just to add `environment` as an optional field, and do renaming in-class. Then remove the `task_params` field once kpet-db has transitioned. This way you won't have to copy the whole schema either.

Reply:
I was planning to remove the whole Succession schema usage from here once kpet-db changes are done. Is that sufficient? I'd like to take this approach.

Reply:
Sure, I understood what you were going to do. Just thought copy-pasting large structures like this gets tedious and error-prone fast and we might have a better approach. But it's OK as is for now.